CMSgov/mint-app
# [NOREF] Added SQLFluff linter

# NOREF

## Changes and Description

- Adds and runs `sqlfluff` linter on all `.sql` files

It's worthy of noting that `sqlfluff` is in beta, and has trouble parsing some statements (e.g. `UPDATE...RETURNING` statements).

It should still do a decent job catching most things in most of our SQL, though.

## How to test this change

```bash
echo 'I should also look over the modified .sql files to make sure nothing broke!'
echo 'I can also modify some .sql files with things that should break a linter, and see if it breaks!'
scripts/dev lint
```

![image](https://user-images.githubusercontent.com/15203744/165533321-30708235-2a80-4cb0-bb79-abc1e5299ca6.png)


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
